pkg/backend/httpstate/backend.go

```diff
@@ -1150,7 +1152,68 @@ func (b *cloudBackend) Preview(ctx context.Context, stack backend.Stack,
 func (b *cloudBackend) Update(ctx context.Context, stack backend.Stack,
 	op backend.UpdateOperation,
 ) (sdkDisplay.ResourceChanges, error) {
-	return backend.PreviewThenPromptThenExecute(ctx, apitype.UpdateUpdate, stack, op, b.apply)
+	return backend.PreviewThenPromptThenExecute(ctx, apitype.UpdateUpdate, stack, op, b.apply, newCopilotExplainer(b))
+}
+
+func (b *cloudBackend) isCopilotFeatureEnabled(projectName tokens.PackageName, opts display.Options) bool {
+	// Have copilot features been requested by specifying the --copilot flag to the cli
+	if !opts.ShowCopilotFeatures {
+		return false
+	}
+
+	// Is copilot enabled this project in Pulumi Cloud
+	contract.Assertf(b.copilotEnabledForCurrentProject != nil,
```

Comment:
Are we 100% sure this will never fail in practice? It makes me slightly nervous - should we return false instead?